### PR TITLE
feat(core): tabs migration

### LINKED ISSUE #8367: Enhancement: Migrate Tabs component to use Overflow Layout
#### Is this a bug, enhancement, or feature request?
enhancement
#### Briefly describe your proposal.
Currently Tabs component uses own implementation of overflow layout functionality.
This task is aimed on standartization of such components so they would use Overflow Layout component.

### PR DESCRIPTION
**Pull request by N1XUS**
## Related Issue(s)

closes #8367 

## Description
Migrated Core Tabs component to use Overflow Layout

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [x] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [x] poor examples
-   [x] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [x] Run npm run build-pack-library and test in external application
-   [x] update `README.md`
-   [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

File: libs/core/src/lib/overflow-layout/directives/overflow-layout-popover-content.directive.ts [hunk 1]
```diff
@@ -21,16 +21,21 @@ export class OverflowLayoutPopoverContentDirective implements OverflowPopoverCon
...
+        // Need to set items with a delay so that elementRef of the focusable item would refresh.
+        setTimeout(() => {
+            this._items = value;
+            this._keyboardEventsManager = new FocusKeyManager(
+                this._items
+                    .filter((item) => item.overflowItem.focusableItem?.focusable)
+                    .map((item) => item.overflowItem.focusableItem)
+            )
+                .withWrap()
+                .withHorizontalOrientation('ltr');
```
Comment (platon-rov) [inline libs/core/src/lib/overflow-layout/directives/overflow-layout-popover-content.directive.ts:33]: no rtf support?

File: libs/core/src/lib/tabs/tab-list.component.html [hunk 2]
```diff
@@ -60,15 +75,35 @@
...
+            >
+            </fd-item-expand>
+            <fd-menu #menu>
+                <li
+                    fd-menu-item
+                    *ngFor="let tab of tabs"
+                    (onSelect)="_overflowingTabHeaderClickHandler(tab.item.panel)"
+                >
+                    <div fd-menu-interactive>
+                        <!--                        <ng-container *ngTemplateOutlet="tab.templateRef"></ng-container>-->
```
Comment (platon-rov) [inline libs/core/src/lib/tabs/tab-list.component.html:93]: probably forgot to remove.

File: libs/core/src/lib/tabs/tab-list.component.ts [hunk 3]
```diff
@@ -88,8 +88,14 @@ export class TabListComponent implements AfterContentInit, AfterViewInit, OnDest
     @Output()
     selectedTabChange = new EventEmitter<TabPanelComponent>();
 
+    @Output()
```
Comment (platon-rov) [inline libs/core/src/lib/tabs/tab-list.component.ts:91]: typedocs.

**Review by platon-rov — COMMENTED** [review]
nice!

**Review by platon-rov — COMMENTED** [review]
seems like these issues were present before so I'm not sure, we can fix it in separate pr probably

Screenshot 2022-08-04 at 15 39 42

Screenshot 2022-08-04 at 15 40 20

**Comment by N1XUS** [comment]
> seems like these issues were present before so I'm not sure, we can fix it in separate pr probably
> 
> Screenshot 2022-08-04 at 15 39 42
> 
> Screenshot 2022-08-04 at 15 40 20

first one could be a separate issue
Second one: it's a tab-nav which does not include overflow functionality since it behaves as a directive.

**Review by platon-rov — APPROVED** [review]

**Review by g-cheishvili — CHANGES_REQUESTED** [review]

**Review by g-cheishvili — APPROVED** [review]

**Review by mikerodonnell89 — DISMISSED** [review]
Keyboard arrow support has changed, previously it was on keydown and holding the arrow would quickly focus through the all of the tabs. Now it is on keyup and doesn't move through them. Is this intentional? Just feels a little sluggish now. Maybe a question not for us but for the designers

**Pull request merged**